lib/Sema/TypeCheckExpr.cpp

```diff
@@ -788,7 +788,8 @@ Expr *CallerSideDefaultArgExprRequest::evaluate(
   auto paramTy = defaultExpr->getType();
 
   // Re-create the default argument using the location info of the call site.
-  auto *initExpr = synthesizeCallerSideDefault(param, defaultExpr->getLoc());
+  auto *initExpr =
+      synthesizeCallerSideDefault(param, defaultExpr->getArgumentListLoc());
```

Comment:
@xedin this is the bit I was missing that caused the test failures previously. `getLoc()` was based on `getSourceRange()`.